src/components/checkbox/checkbox.ts

```diff
@@ -131,11 +134,20 @@ export class MdCheckbox implements ControlValueAccessor {
   }
 
   set checked(checked: boolean) {
-    this._indeterminate = false;
-    this._checked = checked;
-    this._transitionCheckState(
-        this._checked ? TransitionCheckState.Checked : TransitionCheckState.Unchecked);
-    this.change.emit(this._checked);
+    let wasAlreadyInitialized = this._isInitialized;
+    this._isInitialized = true;
+
+    if (checked != this.checked) {
+      this._indeterminate = false;
+      this._checked = checked;
+      this._transitionCheckState(
+          this._checked ? TransitionCheckState.Checked : TransitionCheckState.Unchecked);
+
+      // Only fire a change event if this isn't the first time the checked property is ever set.
+      if (wasAlreadyInitialized) {
```

Comment:
Nitpick: It seems like we could remove the `wasAlreadyInitialized` variable if we move the setting of `this._isInitialized = true`  to the end of the method. e.g.

```typescript
if (this._isInitialized) {
   this.change.emit(this._checked);
}
this._isInitialized = true;
```

What do you think?